src/Cli/dotnet/commands/dotnet-build/BuildCommandParser.cs

```diff
@@ -76,6 +76,8 @@ private static Command ConstructCommand()
             command.AddOption(CommonOptions.ArchitectureOption);
             command.AddOption(CommonOptions.OperatingSystemOption);
 
+            command.TreatUnmatchedTokensAsErrors = false;
```

Comment:
I went through each command, and anytime I found commands that needed to accept arbitrary tokens to forward to another binary, I added this flag.